pygac/gac_io.py

```diff
@@ -75,114 +73,125 @@
+                 'info) = ' + str(first_valid_lat))
+    if end_line > last_valid_lat:
+        LOG.info('New end_line chosen (due to invalid lat/lon '
+                 'info) = ' + str(last_valid_lat))
+
+    # Check user-defined scanlines
+    start_line, end_line = check_user_scanlines(
+        start_line=start_line,
+        end_line=end_line,
+        first_valid_lat=first_valid_lat,
+        last_valid_lat=last_valid_lat)
 
-    # Choose new temporary start/end lines if lat/lon info is invalid
-    no_wrong_lat = np.where(lats != MISSING_DATA_LATLON)
-    temp_start_line = min(no_wrong_lat[0])
-    temp_end_line = max(no_wrong_lat[0])
-    if temp_start_line > start_line:
-        LOG.info('New start_line chosen (due to invalid lat/lon info) = ' + str(temp_start_line))
-    if end_line > temp_end_line:
-        LOG.info('New end_line chosen (due to invalid lat/lon info) = ' + str(temp_end_line))
-
-    # Slice data using temporary start/end lines
-    ref1 = ref1[temp_start_line:temp_end_line + 1, :].copy()
-    ref2 = ref2[temp_start_line:temp_end_line + 1, :].copy()
-    ref3 = ref3[temp_start_line:temp_end_line + 1, :].copy()
-    bt3 = bt3[temp_start_line:temp_end_line + 1, :].copy()
-    bt4 = bt4[temp_start_line:temp_end_line + 1, :].copy()
-    bt5 = bt5[temp_start_line:temp_end_line + 1, :].copy()
-    sun_zen = sun_zen[temp_start_line:temp_end_line + 1, :].copy()
-    sun_azi = sun_azi[temp_start_line:temp_end_line + 1, :].copy()
-    sat_zen = sat_zen[temp_start_line:temp_end_line + 1, :].copy()
-    sat_azi = sat_azi[temp_start_line:temp_end_line + 1, :].copy()
-    rel_azi = rel_azi[temp_start_line:temp_end_line + 1, :].copy()
-    lats = lats[temp_start_line:temp_end_line + 1, :].copy()
-    lons = lons[temp_start_line:temp_end_line + 1, :].copy()
-    miss_lines = np.sort(np.array(
-        qual_flags[0:temp_start_line, 0].tolist() +
-        miss_lines.tolist() +
-        qual_flags[temp_end_line+1:, 0].tolist()
-    ))
-    qual_flags = qual_flags[temp_start_line:temp_end_line+1, :].copy()
-    xutcs = xutcs[temp_start_line:temp_end_line+1].copy()
-
-    # Update user start/end lines to the new slice
-    start_line, end_line = update_start_end_line(
+    # Slice data using new start/end lines
+    _, miss_lines, midnight_scanline = slice_channel(
+        np.zeros(lats.shape),
         start_line=start_line,
         end_line=end_line,
-        temp_start_line=temp_start_line,
-        temp_end_line=temp_end_line)
+        first_valid_lat=first_valid_lat,
+        last_valid_lat=last_valid_lat,
+        qual_flags=qual_flags,
+        miss_lines=miss_lines,
+        midnight_scanline=midnight_scanline)
+
+    ref1, _, _ = slice_channel(ref1,
+                               start_line=start_line,
+                               end_line=end_line,
+                               first_valid_lat=first_valid_lat,
+                               last_valid_lat=last_valid_lat)
```

Comment:
The same operation is performed for all datasets with exactly the same arguments except the first. Could we have a helper function to factorize this ? like:
```python
def slice_channel_inplace(arr, ...):
    arr[:], _, _ = slice_channel(arr, ...)
```
and then just a `for` loop on all the datasets.

Reply:
Absolutely! Will change that

Reply:
Didn't work out because the sliced array is shorter than `arr`. The alternative
```python
sliced = slice_channel()
arr.resize(sliced.shape)
arr[:] = sliced
```
only works for contiguous arrays.